c-layer/lib.cxx

```diff
@@ -27,12 +27,29 @@
 #include <zypp/ui/Selectable.h>
 
 extern "C" {
-static zypp::ZYpp::Ptr zypp_pointer = NULL;
-static zypp::RepoManager *repo_manager = NULL;
 
-void free_zypp() noexcept {
-  zypp_pointer = NULL; // shared ptr assignment operator will free original pointer
-  delete (repo_manager);
+struct Zypp {
+  zypp::ZYpp::Ptr zypp_pointer;
+  zypp::RepoManager *repo_manager;
+};
+
+static struct Zypp the_zypp {
+  .zypp_pointer = NULL, .repo_manager = NULL,
+};
+
+/*
+struct Zypp *new_zypp() {
```

Comment:
oops, that's a leftover before I decided to have a static `the_zypp`, will remove this